### PR TITLE
Throw an ArgumentException when a delegate is bound to an open generi…

### PR DESCRIPTION
**Pull request by joshpeterson**
…c method.

* This change was made in Unity's old version of Mono at: https://github.com/Unity-Technologies/mono/commit/9e27edaec9a91a75d6ebf0d21654dedcab74653d

### CODE REVIEW

**Comment by marek-safar** [comment]
whitelist

**Comment by marek-safar** [comment]
It'd be better to have the test in https://github.com/mono/mono/blob/master/mcs/class/corlib/Test/System/DelegateTest.cs where we have other similar tests

**Comment by joshpeterson** [comment]
@marek-safar: I thought that this would fit better as a runtime test rather than a class library test. But I'm not too familiar with the testing frameworks, so I could be wrong. What do you think?

**Comment by marek-safar** [comment]
We don't have strict rules but in general we write standalone tests only if you cannot easily write nunit test

File: mono/metadata/icall.c [hunk 1]
```diff
@@ -6236,6 +6237,11 @@ ves_icall_System_Delegate_CreateDelegate_internal (MonoReflectionType *type, Mon
 		}
 	}
 
+	if (sig->generic_param_count && method->wrapper_type == MONO_WRAPPER_NONE) {
+		if (!method->is_inflated)
+			mono_raise_exception(mono_get_exception_argument("method", " Cannot bind to the target method because its signature differs from that of the delegate type"));
```
Comment (lambdageek) [inline mono/metadata/icall.c]: Please use `mono_set_pending_exception` and return instead of using `mono_raise_exception`.

Comment (joshpeterson) [inline mono/metadata/icall.c]: Ok, I'll change this. Thanks!

**Comment by joshpeterson** [comment]
@marek-safar: Ok, I'll try to write an nunit test instead.

**Comment by joshpeterson** [comment]
I've updated the code based on the comments here. I noticed that a few of the tests failed, but the failures seemed unrelated. Should I be concerned about these failures, or are they known issues?

**Comment by marek-safar** [comment]
@lambdageek could you review the change

**Comment by lambdageek** [comment]
LGTM.

@monojenkins merge

**Comment by joshpeterson** [comment]
Thanks all, I appreciate the help!

**Pull request merged**